kite-morphlines/kite-morphlines-core/src/main/java/org/kitesdk/morphline/stdlib/HashDigestBuilder.java

```diff
@@ -0,0 +1,215 @@
+    private static final String HASH_TYPE = "hashType";
+    private static final String PRESERVE_EXISTING_NAME = "preserveExisting";
+    private static final boolean PRESERVE_EXISTING_DEFAULT = true;
+
+    public HashDigest(CommandBuilder builder, Config config, Command parent,
+        Command child, MorphlineContext context) {
+      super(builder, config, parent, child, context);
+
+      this.inputFieldName = getConfigs().getString(config, INPUT_FIELD);
+      this.outputFieldName = getConfigs().getString(config, OUTPUT_FIELD);
+      this.hashType = getConfigs().getString(config, HASH_TYPE);
+      this.preserveExisting = getConfigs().getBoolean(config,
+          PRESERVE_EXISTING_NAME, PRESERVE_EXISTING_DEFAULT);
+
+      validateArguments();
+
+      if (LOG.isTraceEnabled()) {
+        LOG.trace("inputField: {}", inputFieldName);
+        LOG.trace("outputField: {}", outputFieldName);
+        LOG.trace("hashType: {}", hashType);
+        LOG.trace("preserveExisting: {}", preserveExisting );
+      }
+    }
+
+    @Override
+    protected boolean doProcess(Record record) {
+
+      if (!preserveExisting) {
+        record.removeAll(outputFieldName);
+      }
+
+      ListIterator iter = record.get(inputFieldName).listIterator();
+      
+      while (iter.hasNext()) {
+        record.put(outputFieldName, doHash(iter.next().toString()));
+      }
+
+      // pass record to next command in chain:
+      return super.doProcess(record);
+
+    }
+
+    private String doHash(String inputString) {
+      MessageDigest digest = DigestUtils.getDigest(hashType);
+
+      byte[] inputBytes = StringUtils.getBytesUtf8(inputString);
+
+      return Hex.encodeHexString(digest.digest(inputBytes));
+    }
+
+    @Override
+    protected void doNotify(Record notification) {
+      super.doNotify(notification);
+    }
+
+  }
+  
+  
+  //Manually shaded from org.apache.commons.codec 1.10.
+  public static class DigestUtils {
+  
+    /**
+     * Returns a <code>MessageDigest</code> for the given <code>algorithm</code>.
+     *
+     * @param algorithm
+     *            the name of the algorithm requested. See <a
+     *            href="http://docs.oracle.com/javase/6/docs/technotes/guides/security/crypto/CryptoSpec.html#AppA"
+     *            >Appendix A in the Java Cryptography Architecture Reference Guide</a> for information about standard
+     *            algorithm names.
+     * @return A digest instance.
+     * @see MessageDigest#getInstance(String)
+     * @throws IllegalArgumentException
+     *             when a {@link NoSuchAlgorithmException} is caught.
+     */
+    public static MessageDigest getDigest(final String algorithm) {
+        try {
+            return MessageDigest.getInstance(algorithm);
+        } catch (final NoSuchAlgorithmException e) {
+            throw new IllegalArgumentException(e);
+        }
+    }
+  }
+
+  //Manually shaded from org.apache.commons.codec 1.8.
+  public static class Hex {
```

Comment:
I'd thought that MessageDigest wasn't re-usable, but having researched that is not the case.